docs/book/src/clusterctl/provider-contract.md

```diff
@@ -150,8 +150,32 @@ will look for objects to reconcile.
 
 #### Variables
 
-The components YAML can contain environment variables matching the regexp `\${\s*([A-Z0-9_]+)\s*}`; it is highly
-recommended to prefix the variable name with the provider name e.g. `${ AWS_CREDENTIALS }`
+The components YAML can contain environment variables matching the format ${VAR}; it is highly
+recommended to prefix the variable name with the provider name e.g. `${AWS_CREDENTIALS}`
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+`clusterctl` currently supports variables with leading/trailing spaces such
+as: `${ VAR }`, `${ VAR}`,`${VAR }`. However, these formats will be deprecated
+in the near future. e.g. v1alpha4.
+
+Formats such as `${VAR$FOO}` is not supported.
```

Comment:
Isn't this supported now? @CecileRobertMichon @wfernandes

Reply:
Good point, it is

Reply:
This is still not supported. There has to be an expression character like `=`, `:=`, `:-" between the variables. This will result in an error.